omigami/utils.py

```diff
@@ -48,3 +50,65 @@ def get_best_n_features(ranks: FeatureRanks, n_to_keep: int) -> List[int]:
                 return feats
 
     raise ValueError("Impossible to return so many best features")
+
+
+def compute_t_student_p_value(sample: float, population: Iterable) -> float:
+    """From Wikipeida:
+    https://en.wikipedia.org/wiki/Prediction_interval#Unknown_mean,_unknown_variance
+    Compute the p_value of the t-Student variable that represent the distribution
+    of the n+1 sample of population, where `n = len(population)`.
+    In few words, it gives the probability that `sample` belongs to `population`.
+    The underlying assumption is that population is normally distributed
+    with unknown mean and unknown variance
+
+    Parameters
+    ----------
+        sample: float
+            value for which we want the p-value
+        population: Iterable
+            values that represent the null hypothesis for `sample`
+
+    Returns
+    -------
+        float
+            the t-Student test p-value
+    """
+    m = np.mean(population)
+    s = np.std(population)
+    n = len(population)
+    t = (sample - m) / (s * (1 + 1 / n) ** 0.5)
+    return scipy.stats.t(df=n - 1).cdf(t)
+
+
+def mute_loggers(loggers: List[str] = None):
+    """Decorator to mute specific loggers within a function call. This is particularly
+    useful when nested for loops might add excessive verbosity to a specific method.
+    The input `loggers` is a list of loggers by name that should be muted. The logging
+    level of the selected loggers is set to WARNING and restored to the original value
+    at the end of the decorated method execution.
+
+    Parameters
+    ----------
+    loggers : List[str], optional
+        List of loggers by name that have to be muted, by default None
+    """
+    loggers = loggers or []
+
+    def decorator(function: Callable):
+        def decorated(*args, **kwargs):
+            logs = {log_name: logging.getLogger(log_name) for log_name in loggers}
+            old_levels = {
+                log_name: log.getEffectiveLevel() for log_name, log in logs.items()
```

Comment:
Yes, it works just as you said. Also regarding the fact that setting logs to warning will make them appear (of course if they are warnings or higher)

I set it to warning because I thought it might still be important to see warnings, but we could set it to ERROR, or add another optional parameter to the decorator to set the in-function level

This is how it looks now
![immagine](https://user-images.githubusercontent.com/66323188/105007905-da2d4380-5a38-11eb-92ea-0563ea0e9ad2.png)

Reply:
Ok I understood it now, thanks